chatgpt-plugin/src/pages/api/handlers/solana-pay/sign/createMerkleTree.ts

```diff
@@ -0,0 +1,70 @@
+import { NextApiRequest } from "next";
+import { Keypair, Connection, PublicKey, Transaction, sendAndConfirmTransaction } from "@solana/web3.js";
+import {
+    ValidDepthSizePair,
+    createAllocTreeIx,
+    SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+    SPL_NOOP_PROGRAM_ID
+} from "@solana/spl-account-compression";
+import { CONNECTION } from "../../../constants";
+import {
+    PROGRAM_ID as BUBBLEGUM_PROGRAM_ID,
+    createCreateTreeInstruction,
+} from "@metaplex-foundation/mpl-bubblegum";
+import dotenv from "dotenv";
+dotenv.config();
+
+export interface MerkleTreeArgs {
+    maxDepth: number
+    maxBufferSize: number
+    canopyHeight: number
+}
+
+export async function createTree(req: NextApiRequest) {
+    const {account: payerPublicKey, canopyDepth, maxDepthSizePair } = req.body;
+    const treeKeypair = Keypair.generate();
+
+    const [treeAuthority, _bump] = PublicKey.findProgramAddressSync(
+        [treeKeypair.publicKey.toBuffer()],
+        BUBBLEGUM_PROGRAM_ID,
+    )
+
+    // instruction for space allocation for tree account
+    const allocTreeIx = await createAllocTreeIx(
+        CONNECTION,
+        treeKeypair.publicKey,
+        payerPublicKey,
+        maxDepthSizePair,
+        canopyDepth
+    )
+
+    // instruction for tree creation
+    const createTreeIx = createCreateTreeInstruction(
+        {
+            payer: payerPublicKey,
+            treeCreator: payerPublicKey,
+            treeAuthority: treeAuthority,
+            merkleTree: treeKeypair.publicKey,
+            compressionProgram: SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+            logWrapper: SPL_NOOP_PROGRAM_ID
+        },
+        {
+            maxBufferSize: maxDepthSizePair.maxBufferSize,
+            maxDepth: maxDepthSizePair.maxDepth,
+            public: false,
+        },
+        BUBBLEGUM_PROGRAM_ID
+    )
+
+    try {
+        const tx = new Transaction();
+        tx.add(allocTreeIx);
+        tx.add(createTreeIx);
+        tx.feePayer = payerPublicKey;
+        return tx;
+
+    } catch (e) {
+        console.error(e);
+        throw e;
+    }
```

Comment:
based off the missing `tx.serialize({})` call I'm assuming this is untested? it should be possible to add a `closeTree` ix as well.

Please add a comment at the top of this file saying to come back around and add a `closeTree` ix before exposing this endpoint. We don't want users paying 20 sol for tree & be stuck because there's no way to get their sol back.